src/test/java/uk/gov/hmcts/reform/managecase/service/noc/VerifyNoCAnswersServiceTest.java

```diff
@@ -0,0 +1,160 @@
+package uk.gov.hmcts.reform.managecase.service.noc;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import uk.gov.hmcts.reform.managecase.api.payload.VerifyNoCAnswersRequest;
+import uk.gov.hmcts.reform.managecase.client.datastore.model.CaseViewResource;
+import uk.gov.hmcts.reform.managecase.client.datastore.model.elasticsearch.SearchResultViewItem;
+import uk.gov.hmcts.reform.managecase.client.definitionstore.model.ChallengeQuestionsResult;
+import uk.gov.hmcts.reform.managecase.client.prd.FindUsersByOrganisationResponse;
+import uk.gov.hmcts.reform.managecase.domain.NoCRequestDetails;
+import uk.gov.hmcts.reform.managecase.repository.PrdRepository;
+import uk.gov.hmcts.reform.managecase.service.NoticeOfChangeService;
+
+import javax.validation.ValidationException;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.when;
+
+@SuppressWarnings({"PMD.JUnitAssertionsShouldIncludeMessage", "PMD.DataflowAnomalyAnalysis",
+    "PMD.UseConcurrentHashMap"})
+class VerifyNoCAnswersServiceTest {
+
+    @InjectMocks
+    private VerifyNoCAnswersService verifyNoCAnswersService;
+
+    @Mock
+    private NoticeOfChangeService noticeOfChangeService;
+
+    @Mock
+    private ChallengeAnswerValidator challengeAnswerValidator;
+
+    @Mock
+    private PrdRepository prdRepository;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private SearchResultViewItem searchResultViewItem;
+    private ChallengeQuestionsResult challengeQuestionsResult;
+    private CaseViewResource caseViewResource;
+
+    @BeforeEach
+    void setUp() throws JsonProcessingException {
+        MockitoAnnotations.initMocks(this);
+
+        searchResultViewItem = createCase();
+        challengeQuestionsResult = new ChallengeQuestionsResult();
+        caseViewResource = new CaseViewResource();
+        NoCRequestDetails details = NoCRequestDetails.builder()
+            .searchResultViewItem(searchResultViewItem)
+            .challengeQuestionsResult(challengeQuestionsResult)
+            .caseViewResource(caseViewResource)
+            .build();
+
+        when(noticeOfChangeService.challengeQuestions("1")).thenReturn(details);
+    }
+
+    @Test
+    void shouldVerifyNoCAnswersSuccessfully() {
+        mockPrdResponse("ORGID1");
+        when(challengeAnswerValidator.getMatchingCaseRole(Mockito.any(), Mockito.any(), Mockito.any()))
```

Comment:
use expected values instead of `any` ?

Reply:
Since there is no real logic for those values, for the purpose of the test I don't think it adds anything.

Reply:
👍